### PR TITLE
Added Windows API Contract Checker

### LINKED ISSUE #1289: Add logic to know which SDK version we're targeting to better handle backwards comparability
Right now, Motion Controller visualizer depends on 16299 SDK calls ~~that will prevent us from building for SDK targets less than 16299.~~

HoloLens OS is still ~~15063~~ 14393.

Update
We will update the min SDK target to 16299 and update the scripts in the toolkit with `Windows.Foundation.Metadata.ApiInformation` API contract checks.

Affected scripts:
- [x] `InteractionExtensions.cs`
- [x] Build And Deploy Window

### LINKED ISSUE #1325: Confusing text around Current supported versions
Maybe this is just a formatting issue, but it feels like there is unnecessary confusion in the text on this page. I am assuming that "Mixed Reality" is the blanket term, and includes both "Windows Mixed Reality" headsets and HoloLens.

    Note: The latest release should work for both HoloLens and Windows Mixed Reality development.
    Latest Master Branch Release v1.2017.2.0
    Current Unity Editor Version 2017.2.0p1 MRTP4

    Check out the updates from the Fall Creators update for Windows Mixed Reality

        HoloLens Developers
        Currently we are waiting on a fix for HoloLens development from Unity, for now you should use the 2017.1.2 version of Unity and the "Release" version of the MRTK Asset:
         Unity Editor HoloLens Version 2017.1.2f1

If the Unity editor version is still different for WMR Headsets, then I think you need another section to call this out (and if it is not then I'm even more confused!):

    All MR Developers (both Windows Mixed Reality and HoloLens)     
    Note: The latest release should work for both HoloLens and Windows Mixed Reality development.
    Latest MR Toolkit Master Branch Release v1.2017.2.0

    Windows Mixed Reality Developers
    Current Unity Editor Version 2017.2.0p1 MRTP4

    HoloLens Developers
    Unity Editor HoloLens Version 2017.1.2f1

Then the text on this page: https://github.com/Microsoft/MixedRealityToolkit-Unity/releases/tag/v1.2017.2.0  still links to the Unity MRTP3 version, rather than the MRTP4 version linked on the previous page.

### PR DESCRIPTION
**Pull request by StephenHodgson**
Fixes #1289
Fixes #1325

- Made the 16299 SDK the default target when building with the Build And Deploy Window.
- Made msbuild 15.0 the default build version number.
- Update readme documentation with editor usage for Hololens/IMHD.
    - After this PR is merged we should be able to use one editor version for all WMR apps.

### CODE REVIEW

File: Assets/HoloToolkit-Examples/Utilities/Scripts/CheckApiTest.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+using UnityEngine;
+using UnityEngine.UI;
+
+public class CheckApiTest : MonoBehaviour
+{
+    [SerializeField]
+    private Text text;
+
+    public void CheckV5Api()
```
Comment (keveleigh) [inline Assets/HoloToolkit-Examples/Utilities/Scripts/CheckApiTest.cs]: Pressing the buttons with a motion controller doesn't seem to call this code (voice "select" works). Is that something that'll be fixed with your uGUI fixes?

File: Assets/HoloToolkit-Examples/Utilities/Scenes/CheckAPI.unity [hunk 2]
```diff
@@ -0,0 +1,1180 @@
...
+  serializedVersion: 5
+  m_Component:
+  - component: {fileID: 1598773018}
+  - component: {fileID: 1598773017}
+  - component: {fileID: 1598773016}
+  - component: {fileID: 1598773015}
+  - component: {fileID: 1598773014}
+  - component: {fileID: 1598773019}
+  m_Layer: 5
+  m_Name: Canvas
```
Comment (keveleigh) [inline Assets/HoloToolkit-Examples/Utilities/Scenes/CheckAPI.unity]: The buttons are in the floor on an immersive HMD, so we should probably put a `SceneContentAdjuster` on them.
Also, I tried that, and the `SceneContentAdjuster` only seems to work in Play mode. When I built and deployed to an immersive HMD, the buttons were still in the floor. Was that tested to work when deployed?

Comment (keveleigh) [reply Assets/HoloToolkit-Examples/Utilities/Scenes/CheckAPI.unity]: `SceneContentAdjuster` also overwrites the X and Z values, which doesn't seem ideal.

File: Assets/HoloToolkit-Examples/Utilities/Scripts/CheckApiTest.cs (hunk 1 above)
Comment (StephenHodgson) [reply Assets/HoloToolkit-Examples/Utilities/Scripts/CheckApiTest.cs]: That is correct. Will be fixed by the uGUI raycast PR

File: Assets/HoloToolkit-Examples/Utilities/Scenes/CheckAPI.unity (hunk 2 above)
Comment (StephenHodgson) [reply Assets/HoloToolkit-Examples/Utilities/Scenes/CheckAPI.unity]: I'll double check this in a different PR. Open an issue for me please?

Should be working when deployed (Worked for me in MRTP3, but they did change the way that stuff works so idk)

File: Assets/HoloToolkit/Utilities/Scripts/Extensions/InteractionSourceExtensions.cs [hunk 3]
```diff
@@ -89,19 +99,22 @@ public static void StopHaptics(this InteractionSource interactionSource)
         public static IAsyncOperation<IRandomAccessStreamWithContentType> TryGetRenderableModelAsync(this InteractionSource interactionSource)
         {
             IAsyncOperation<IRandomAccessStreamWithContentType> returnValue = null;
-
-            UnityEngine.WSA.Application.InvokeOnUIThread(() =>
+
```
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/HoloToolkit/Utilities/Scripts/Extensions/InteractionSourceExtensions.cs]: What's wrong with the whitespace here?

[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/HoloToolkit/Utilities/Scripts/Extensions/InteractionSourceExtensions.cs]: nvm I see it, thanks!

File: Assets/HoloToolkit-Examples/Utilities/Scenes/CheckAPI.unity (hunk 2 above)
Comment (StephenHodgson) [reply Assets/HoloToolkit-Examples/Utilities/Scenes/CheckAPI.unity]: @keveleigh I'll address this in #1309 Removed OptimizeSceneForDeviceType

**Pull request merged**